source/fundamentals/crud/write-operations/embedded-arrays.txt

```diff
@@ -0,0 +1,283 @@
+```````
+
+The following example performs the following actions:
+
+- Query for embedded arrays where "B" exists in warehouses
+- Use the ``inc()`` method of the ``Updates`` builder to decrement the ``qty`` field by "3" 
+- Use a positional ``$`` operator to indicate an update to the first matched embedded array
+
+.. literalinclude:: /includes/fundamentals/code-snippets/UpdateArray.java
+   :language: java
+   :dedent:
+   :start-after: begin updateValueExample
+   :end-before: end updateValueExample
+
+The output of the above code should look like this:
+
+.. code-block:: json
+   :copyable: false
+
+    { "_id": 1, 
+      "color": "green", 
+      "instock": [
+        { "qty": 8, "warehouse": ["A", "E"] }, 
+        { "qty": 10, "warehouse": ["B", "C"] }, 
+        { "qty": 15, "warehouse": ["B", "F"] } ]}
+
+For additional information on the operator and method mentioned in this
+section, see the following resources: 
+
+- :manual:`Positional $ Operator </reference/operator/update/positional>` Server Manual Explanation
+- :java-core-api:`inc() <com/mongodb/client/model/Updates.html#inc(java.lang.String,java.lang.Number)>` API Documentation
+
+All Array Elements
+~~~~~~~~~~~~~~~~~~
+
+Use the all positional ``$[]`` operator to update each embedded array in
+the document. 
+
+Example
+```````
+
+The following example performs the following actions: 
+
+- Using the ``inc()`` method of the ``Updates`` builder to increment the ``qty`` field by "5"
+- Use an all positional operator to indicate an update to each embedded array
+
+.. literalinclude:: /includes/fundamentals/code-snippets/UpdateArray.java
+   :language: java
+   :dedent:
+   :start-after: begin updateAllElementsExample
+   :end-before: end updateAllElementsExample
+
+The example above updates the original document to the following state:
+
+.. code-block:: json
+   :copyable: false
+
+    { "_id": 1, 
+      "color": "green", 
+      "instock": [
+        {"qty": 13, "warehouse": ["A", "E"]}, 
+        {"qty": 18, "warehouse": ["B", "C"]}, 
+        {"qty": 20, "warehouse": ["B", "F"]} ]}
+
+For additional information on the operator and method mentioned in this
+section, see the following resources: 
+
+- :manual:`Positional $[] Operator </reference/operator/update/positional-all/>` Server Manual Explanation
+- :java-core-api:`inc() <com/mongodb/client/model/Updates.html#inc(java.lang.String,java.lang.Number)>` API Documentation
+
+.. _filtered_positional_operator:
+
+Filtered Array Elements
+~~~~~~~~~~~~~~~~~~~~~~~
+
+Specify a ``UpdateOptions.arrayFilters`` condition and use the filtered
+positional ``$[<identifier>]`` operator to update certain embedded
+arrays in a document. 
+
+The ``<identifier>`` is the placeholder value you give to represent an
+element of the array field. This value is the prefix of the array
+variable in your ``arrayFilters``. 
+
+.. note::
+
+    The ``<identifier>`` must begin with a lowercase letter and contain
+    only alphanumeric characters.
```

Comment:
```suggestion
variable in your ``arrayFilters``.  The ``<identifier>`` must begin with a lowercase letter and contain
only alphanumeric characters.
```

Question: Is there a reason why this is in a separate note?

Reply:
removed the note and put the content within the paragraph